encoding/graph/type_/vertex_generator.rs

```diff
@@ -4,27 +4,85 @@
  * file, You can obtain one at https://mozilla.org/MPL/2.0/.
  */
 
-use std::sync::atomic::{AtomicU16, Ordering};
+use std::sync::atomic::AtomicU16;
+use std::sync::atomic::Ordering::Relaxed;
+use bytes::Bytes;
+use primitive::prefix_range::PrefixRange;
+use resource::constants::snapshot::BUFFER_KEY_INLINE;
 
-use storage::{snapshot::WriteSnapshot, MVCCStorage};
+use storage::key_value::{StorageKey, StorageKeyReference};
 use storage::snapshot::WritableSnapshot;
 
 use crate::{
     graph::type_::vertex::{
-        build_vertex_attribute_type, build_vertex_attribute_type_prefix, build_vertex_entity_type,
-        build_vertex_entity_type_prefix, build_vertex_relation_type, build_vertex_relation_type_prefix,
-        build_vertex_role_type, build_vertex_role_type_prefix, TypeID, TypeVertex,
+        build_vertex_attribute_type, build_vertex_entity_type, build_vertex_relation_type,
+        build_vertex_role_type, TypeID, TypeVertex,
```

Comment:
These are currently generated by macros, though they could likely be const instances of a struct.

```rust
pub struct TypeVertexConstructor {
    prefix : Prefix
}

impl TypeVertexConstructor {
    pub fn new_vertex(&self, bytes: Bytes<'_, BUFFER_KEY_INLINE>) -> TypeVertex<'_> {
        let vertex = TypeVertex::new(bytes);
        debug_assert_eq!(vertex.prefix(), self.prefix);
        vertex
    }

    pub fn build_vertex(&self, type_id: TypeID) -> TypeVertex<'static> {
        TypeVertex::build(self.prefix.prefix_id(), type_id)
    }

    pub fn build_prefix(&self) -> StorageKey<'static, { TypeVertex::LENGTH_PREFIX }> {
        StorageKey::new_ref(TypeVertex::KEYSPACE, ByteReference::new(&self.prefix.prefix_id().bytes))
    }

    pub fn is_same_kind(&self, bytes: Bytes<'_, BUFFER_KEY_INLINE>) -> bool {
        bytes.length() == TypeVertex::LENGTH && TypeVertex::new(bytes).prefix() == self.prefix
    }
}

const ENTITY_TYPE_CONSTRUCTOR : TypeVertexConstructor = TypeVertexConstructor { prefix: Prefix::VertexEntityType };
const RELATION_TYPE_CONSTRUCTOR : TypeVertexConstructor = TypeVertexConstructor { prefix: Prefix::VertexRelationType };
const ATTRIBUTE_TYPE_CONSTRUCTOR : TypeVertexConstructor = TypeVertexConstructor { prefix: Prefix::VertexAttributeType };
const ROLE_TYPE_CONSTRUCTOR : TypeVertexConstructor = TypeVertexConstructor { prefix: Prefix::VertexRoleType };
```

Reply:
Added a todo in the code to do it.